### PR TITLE
[release-3.11] Including fluent-plugin-concat to support cri-o multiline logs.

### PR DESCRIPTION
**Pull request by nhosoi**
Backporting the partial code in fluent-plugin-concat v2.3.0 to v1.0.0.
Note: removing @seperator = '\n' from join.  It adds '\n' at the end of
      the partial message.  The original code has the issue.
  - @key => lines.join(@separator)
  + @key => lines.join

Added missing keep_partial_key code.
Moved filter_concat.rb to lib/filter_concat and added a unit test.
Fixed broken unit test in test/generate_throttle_configs_test.rb.
Enabling filter_concat unit test.
Capturing exception in shutdown.

(cherry picked from commit 30efccb7a7bebbdb09193248934186f54c4294f2)

### CODE REVIEW

**Comment by richm** [comment]
updated aos cd jobs
/retest

**Comment by richm** [comment]
If we need this in 3.11, you'll need to change it to use the `use_crio` flag - the automatic detection of crio or json-file was not backported to 3.11

**Comment by nhosoi** [comment]
> If we need this in 3.11, you'll need to change it to use the use_crio flag - the automatic detection of crio or json-file was not backported to 3.11

Yes, that's what I manually backported pr/1350 since /cherrypick release-3.11 failed because of it.

File: fluentd/lib/generate_throttle_configs/lib/generate_throttle_configs.rb [hunk 1]
```diff
@@ -168,13 +168,26 @@ def create_default_docker(input_conf_file, excluded, log, options={})
   @label @INGRESS
   path "#{cont_logs_path}"
   pos_file "#{cont_pos_file}"
-  time_format #{use_crio == 'true' ? '%Y-%m-%dT%H:%M:%S.%N%:z' : '%Y-%m-%dT%H:%M:%S.%N%Z'}
+  time_format '%Y-%m-%dT%H:%M:%S.%N%Z'
```
Comment (richm) [inline fluentd/lib/generate_throttle_configs/lib/generate_throttle_configs.rb]: I don't understand this.  Don't we need this in 3.11 for crio support?

Comment (nhosoi) [reply fluentd/lib/generate_throttle_configs/lib/generate_throttle_configs.rb]: Sorry, I missed the time_format change...  Pushed the fix.

**Comment by richm** [comment]
These problems with kibana not starting are persistent . . .
/retest

**Pull request merged**